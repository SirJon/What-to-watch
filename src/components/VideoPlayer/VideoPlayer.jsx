import React, { PureComponent } from "react";
import PropTypes from 'prop-types';

class VideoPlayer extends PureComponent {
  constructor(props) {
    super(props);
    this.video = React.createRef();
    this.timer
  }
  playVideo(video) {
    this.timer = setTimeout(() => {
      video.play();
    }, 1000)
  };
  stopVideo(video) {
    this.timer = clearTimeout(this.timer);
    video.load();
  };
  render() {
    const { name, img, video, onMouse } = this.props;
    return (
      <article
        className="small-movie-card catalog__movies-card"
        onMouseOver={() => {
          onMouse(name);
          this.playVideo(this.video.current);
        }
        }
        onMouseOut={() => this.stopVideo(this.video.current)}
      >
        <div className="small-movie-card__image">
          <video
            width="280"
            height="175"
            poster={img}
            ref={this.video}
            muted="muted"
          >
            <source src={video} />
          </video>
        </div>
        <h3 className="small-movie-card__title">
          <a
            className="small-movie-card__link"
            href="#"
            onClick={(e) => e.preventDefault()}
          >{name}</a>
        </h3>
      </article>
    )
  }
};

VideoPlayer.propTypes = {
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  video: PropTypes.string.isRequired,
  onMouse: PropTypes.func.isRequired,
};

export default VideoPlayer;
