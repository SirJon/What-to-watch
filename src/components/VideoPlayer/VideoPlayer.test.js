import renderer from 'react-test-renderer';
import React from "react";
import VideoPlayer from './VideoPlayer';

const data = {
  key: `0`,
  name: `1`,
  img: `2`,
  video: `3`,
  onMouse: () => console.log(`4`)
};

describe(`Films`, () => {
  it(`Render`, () => {
    const tree = renderer
      .create(
        <VideoPlayer
          key={data.key}
          name={data.name}
          img={data.img}
          video={data.video}
          onMouse={data.onMouse}
        />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
