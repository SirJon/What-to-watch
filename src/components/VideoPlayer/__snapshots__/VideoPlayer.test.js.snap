// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films Render 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      muted="muted"
      width="280"
    >
      <source />
    </video>
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
      onClick={[Function]}
    />
  </h3>
</article>
`;
